### PR TITLE
TDI-38305:tAzurestorageList flow variable "current blob".

### PR DESCRIPTION
**Pull request by pyzhou-talend**
**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TDI-38305

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/306/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/306/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/308/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/308/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/309/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/309/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/316/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/316/)

File: components/components-azurestorage/src/main/java/org/talend/components/azurestorage/blob/runtime/AzureStorageListReader.java [hunk 1]
```diff
@@ -85,10 +87,16 @@ public boolean start() throws IOException {
...
             blobIndex = 0;
-            currentRecord = new GenericData.Record(properties.schema.schema.getValue());
-            currentRecord.put(0, blobs.get(blobIndex).getName());
+
+            IndexedRecord dataRecord = new GenericData.Record(properties.schema.schema.getValue());
+            dataRecord.put(0, blobs.get(blobIndex).getName());
+            Schema rootSchema = RootSchemaUtils.createRootSchema(properties.schema.schema.getValue(), properties.outOfBandSchema);
+            currentRecord = new GenericData.Record(rootSchema);
+            currentRecord.put(0, dataRecord);
+            currentRecord.put(1, dataRecord);
```
Comment (gonchar-ivan) [inline components/components-azurestorage/src/main/java/org/talend/components/azurestorage/blob/runtime/AzureStorageListReader.java]: Looks strange as '1' index is for outOfBandRecord - record with flow variables values,
but here dataRecord is used 2 times

File: components/components-azurestorage/src/main/java/org/talend/components/azurestorage/blob/tazurestoragelist/TAzureStorageListProperties.java [hunk 2]
```diff
@@ -33,6 +34,17 @@
 
     protected transient PropertyPathConnector MAIN_NAME = new PropertyPathConnector(Connector.MAIN_NAME, "schema");
 
+	public static final Schema outOfBandSchema;
```
Comment (gonchar-ivan) [inline components/components-azurestorage/src/main/java/org/talend/components/azurestorage/blob/tazurestoragelist/TAzureStorageListProperties.java]: This is a constant. (It is static and final)
Maybe better to use OUT_OF_BAND_SCHEMA field name for it?

Comment (pyzhou-talend) [inline components/components-azurestorage/src/main/java/org/talend/components/azurestorage/blob/tazurestoragelist/TAzureStorageListProperties.java]: ok. I will change it in another PR.

**Pull request merged**